### PR TITLE
Movies list

### PR DESCRIPTION
**Pull request by ssorguc**
XML data parsed, scrollbar works, constraints set
screen shot 2017-10-17 at 22 24 43

screen shot 2017-10-17 at 22 27 04

### CODE REVIEW

File: Popular Movies/Popular Movies/Assets.xcassets/movieSample.imageset/Contents.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
+  "images" : [
+    {
+      "idiom" : "universal",
+      "scale" : "1x"
+    },
+    {
+      "idiom" : "universal",
```
Comment (nejrazulic) [inline Popular Movies/Popular Movies/Assets.xcassets/movieSample.imageset/Contents.json:8]: Please, remove these photos you use for testing (Interstellar poster)

File: Popular Movies/Popular Movies/Controllers/MoviesCollectionViewController.m [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+#import "MovieTvShowCollectionViewCell.h"
+@interface MoviesCollectionViewController ()
+{
+    NSMutableArray* moviesArray;
+}
+@end
+
+@implementation MoviesCollectionViewController
+
+static NSString * const reuseIdentifier = @"Cell";
```
Comment (nejrazulic) [inline Popular Movies/Popular Movies/Controllers/MoviesCollectionViewController.m:19]: Identifier should be encapsulated in cell class

File: Popular Movies/Popular Movies/Controllers/MoviesCollectionViewController.m [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+}
+
+- (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView cellForItemAtIndexPath:(NSIndexPath *)indexPath {
+    MovieTvShowCollectionViewCell *cell = [collectionView dequeueReusableCellWithReuseIdentifier:reuseIdentifier forIndexPath:indexPath];
+    return cell;
+}
+
+#pragma mark <UICollectionViewDelegate>
+
+/*
```
Comment (nejrazulic) [inline Popular Movies/Popular Movies/Controllers/MoviesCollectionViewController.m:71]: Unused code should be removed

File: Popular Movies/Popular Movies/Controllers/MoviesCollectionViewController.m (hunk 2 above)
Comment (nejrazulic) [reply Popular Movies/Popular Movies/Controllers/MoviesCollectionViewController.m:19]: Also, use specific names for constants e.g. movieCellReuseIdentifier

File: Popular Movies/Popular Movies/Controllers/RssFeedTableViewController.m [hunk 4]
```diff
@@ -11,28 +11,32 @@
...
-    lista_vijesti = [[NSMutableArray alloc]init];
-    News * news1 = [[News alloc] init];
-    news1.title = @"Music For Self Improvment";
-    news1.source_link = @"www.source.com";
-    news1.text = @"When I was just starting 6th grade I got my first job. Paperboy! Boy, was I excited. At that time I had spent a lot of time actually playing the video game Paperboy, so I knew I had what it took to get the job done. But, its just not that easy.";
-    [lista_vijesti addObject:news1];
-    [lista_vijesti addObject:news1];
-    [lista_vijesti addObject:news1];
-
+    newsArray = [[NSMutableArray alloc]init];
```
Comment (nejrazulic) [inline Popular Movies/Popular Movies/Controllers/RssFeedTableViewController.m:33]: This shouldn't be part of controller. Create NewsServices class and put there methods that will fetch news, so you can load news in one line from viewDidLoad.

File: Popular Movies/Popular Movies/Controllers/RssFeedTableViewController.m [hunk 5]
```diff
@@ -46,26 +50,62 @@ - (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
...
     cell.linkLabel.text = newsSample.source_link;
-    cell.contentTextBox.text = newsSample.text;
+    cell.descriptionLabel.text = newsSample.text;
     return cell;
 }
 -(CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath{
     return 200.0f;
 }
 
+-(void)parser:(NSXMLParser *)parser {
```
Comment (nejrazulic) [inline Popular Movies/Popular Movies/Controllers/RssFeedTableViewController.m:68]: remove unused code

**Review by nejrazulic — CHANGES_REQUESTED** [review]

File: Popular Movies/Popular Movies/Controllers/MoviesCollectionViewController.h [hunk 6]
```diff
@@ -0,0 +1,12 @@
+//
+//  MoviesCollectionViewController.h
+//  Popular Movies
+//
+//  Created by Test on 14/10/2017.
+//  Copyright © 2017 Test. All rights reserved.
+//
+
+#import <UIKit/UIKit.h>
+
```
Comment (adiscehajicabh) [inline Popular Movies/Popular Movies/Controllers/MoviesCollectionViewController.h:10]: **Suggestion:** If you inherit **UICollectionViewController** class it is not needed to implement **UICollectionViewDelegate** and **UICollectionViewDataSource** protocols. This is needed when you inherit **UIViewController** class.

File: Popular Movies/Popular Movies/Controllers/RssFeedTableViewController.h [hunk 7]
```diff
@@ -8,6 +8,5 @@
 
 #import <UIKit/UIKit.h>
```
Comment (adiscehajicabh) [inline Popular Movies/Popular Movies/Controllers/RssFeedTableViewController.h:10]: **Same suggestion:** If you inherit **UITableViewController** class it is not needed to implement **UITableViewDelegate** and **UITableViewDataSource** protocols. This is needed when you inherit **UIViewController** class.

File: Popular Movies/Popular Movies/Model/Date.h [hunk 8]
```diff
@@ -0,0 +1,15 @@
+//
+//  Date.h
+//  Popular Movies
+//
+//  Created by Test on 14/10/2017.
+//  Copyright © 2017 Test. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+
```
Comment (adiscehajicabh) [inline Popular Movies/Popular Movies/Model/Date.h:10]: Is there a specific reason why do we need this class? If we need to use part of the date (e.g. day or month) we can use **[NSDateComponents](https://developer.apple.com/documentation/foundation/nsdatecomponents)**.

**Pull request merged**